src/main/java/com/shapesecurity/salvation/data/Notice.java

```diff
@@ -0,0 +1,76 @@
+package com.shapesecurity.salvation.data;
+
+import com.shapesecurity.salvation.interfaces.Show;
+
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Collectors;
+
+public class Notice implements Show {
+
+    @Nonnull public final Type type;
+    @Nonnull public final String message;
+    @Nullable public Location startLocation;
+    @Nullable public Location endLocation;
+
+
+    public Notice(@Nonnull Type type, @Nonnull String message) {
+        this.message = message;
+        this.type = type;
+    }
+
+    @Override public String toString() {
+        return type.getValue() + ": " + message;
+    }
+
+    @Nonnull
+    public static ArrayList<Notice> getAllErrors(@Nullable ArrayList<Notice> notices) {
+        if (notices == null) {
+            return new ArrayList<>();
+        }
+        return notices.stream().filter(notice -> notice.type == Type.ERROR).collect(Collectors.toCollection(ArrayList::new));
```

Comment:
Add `isError`, `isWarning`, and `isInfo` functions that do `this.type == Type.ERROR`, etc. and then use `Notice::isError` here and similar below.